### PR TITLE
[search] shop-houseware category.

### PR DESCRIPTION
**Pull request by vng**
Added 'houseware' category only for English/Russian. I don't like straightforward translations like de = Haushaltswarengeschäft :) to be used in search terms.

If somebody can advise good short search definitions for shop=houseware in his native language, I will add them.

### CODE REVIEW

**Comment by Misalf-git** [comment]
True, the german language has some long words due to combination.
If it's obvious that this is a store ("Geschäft"), then just "Haushaltswaren" may be sufficient.

File: data/strings/types_strings.txt [hunk 1]
```diff
@@ -23284,13 +23284,13 @@
     pt = Loja de utilidades domésticas
     pt-BR = Loja de utilidades domésticas
     ro = Magazin de articole de uz casnic
-    ru = Магазин посуды
+    ru = Товары для дома (бытовые)
```
Comment (pastk) [inline data/strings/types_strings.txt:23287]: Бытовые товары?

also remove the `ref =` please as its not needed anymore

**Review by pastk — APPROVED** [review]

**Comment by biodranik** [comment]
> I don't like straightforward translations like de = Haushaltswarengeschäft :) to be used in search terms.

It is not important what we think about these terms. Important is _how German users are typing it_. I suggest to enable suggestions in such cases, then long words are not an issue at all.

File: data/strings/types_strings.txt [hunk 2]
```diff
@@ -23260,7 +23260,7 @@
     en-GB = Housewares Shop
     af = Huiswarewinkel
     ar = محل أدوات منزلية
-    be = Крама посуду
+    be = Тавары для дома (бытавыя)
```
Comment (biodranik) [inline data/strings/types_strings.txt:23263]:
```suggestion
    be = Бытавыя тавары
```

File: data/strings/types_strings.txt (hunk 1 above)
Comment (biodranik) [inline data/strings/types_strings.txt:23287]:
```suggestion
    ru = Бытовые товары
```

File: data/strings/types_strings.txt [hunk 3]
```diff
@@ -23284,13 +23284,13 @@
...
     ro = Magazin de articole de uz casnic
-    ru = Магазин посуды
+    ru = Товары для дома (бытовые)
     sk = Obchod s domácimi potrebami
     sv = Husgerådsbutik
     sw = Duka la Vifaa vya Nyumbani
     th = ร้านของใช้ในบ้าน
     tr = Ev Eşyaları Dükkanı
-    uk = Магазин посуду
+    uk = Товари для дому (побутові)
```
Comment (biodranik) [inline data/strings/types_strings.txt:23293]:
```suggestion
    uk = Побутові товари
```

File: indexer/search_string_utils.cpp [hunk 4]
```diff
@@ -130,6 +130,10 @@ UniString NormalizeAndSimplifyString(std::string_view s)
       c = 'a';
       uniString.insert(uniString.begin() + (i++) + 1, 'e');
       break;
+    case 0x2018:  // ‘
+    case 0x2019:  // ’
+      c = '\'';
```
Comment (biodranik) [inline indexer/search_string_utils.cpp:135]: Can we reuse the same constant if it's also used somewhere in the search engine, for clarity, why is it replaced here?

File: data/categories.txt [hunk 5]
```diff
@@ -1614,6 +1614,12 @@ el:Είδη κιγκαλερίας|DIY|Κάν'το μόνος σου|κατάσ
 sk:Železiarstvo|Náradie
 fa:فروشگاه ابزار و یراق
 
+shop-houseware|@shop
+en:4Houseware
+be:4Пасуда|4Бытавыя тавары
+ru:4Посуда|4Бытовые товары
+uk:4Посуд|4Побутові товари
+
```
Comment (jimcarst) [inline data/categories.txt:1622]:
```suggestion
nl: 4Huishouden
```

**Review by jimcarst — APPROVED** [review]

**Review by matheusgomesms — APPROVED** [review]
PT and PT-BR looks fine!

File: data/categories.txt (hunk 5 above)
Comment (biodranik) [reply data/categories.txt:1627]:
```
    en = Housewares
    ar = بضائع منزلية
    be = Бытавыя тавары
    bg = Домакински стоки
    ca = Articles de la llar
    cs = Domácí potřeby
    da = Husholdningsartikler
    de = Haushaltswaren
    el = Οικιακά αγαθά
    es = Enseres domésticos
    es-MX = Artículos de uso doméstico
    et = Kodutarbed
    eu = Etxeko ondasunak
    fa = کالاهای خانگی
    fi = Kotitaloustavarat
    fr = Articles ménagers
    he = מוצרים לבית
    hi = घरेलू सामान
    hu = Háztartási cikkek
    id = Barang-barang rumah tangga
    it = Articoli per la casa
    ja = 家庭用品
    ko = 가정용품
    mr = घरगुती वस्तू
    nb = Husholdningsartikler
    nl = Huishoudelijke artikelen
    pl = Artykuły gospodarstwa domowego
    pt = Bens de uso doméstico
    pt-BR = Bens domésticos
    ro = Bunuri de uz casnic
    ru = Бытовые товары
    sk = Tovar pre domácnosť
    sv = Hushållsartiklar
    sw = Bidhaa za nyumbani
    th = สินค้าใช้ในบ้าน
    tr = Ev eşyaları
    uk = Побутові товари
    vi = Hàng gia dụng
    zh-Hans = 家庭用品
    zh-Hant = 家庭用品
```

**Review by biodranik — APPROVED** [review]
Did you check emoji before removal? Are they wrong?

File: data/categories.txt (hunk 5 above)
Comment (dvdmrtnz) [reply data/categories.txt:1627]: es = Enseres domésticos
es-MX = Enseres domésticos

**Comment by vng** [comment]
Deleted emojis are duplicates from the base _shop_ category

File: data/categories.txt (hunk 5 above)
Comment (vng) [reply data/categories.txt:1627]: So you want to replace type_strings or categories or both?

Comment (biodranik) [reply data/categories.txt:1627]: Do not replace, fill in just missing parts.

**Review by biodranik — APPROVED** [review]

**Pull request merged**